llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020-zeroinitializer-array-of-arrays.ll

```diff
@@ -0,0 +1,123 @@
+attributes #5 = { convergent inlinehint mustprogress norecurse "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" }
+attributes #6 = { convergent norecurse nounwind "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" }
+attributes #7 = { convergent mustprogress norecurse nounwind "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" }
+attributes #8 = { argmemonly nofree nounwind willreturn writeonly }
+attributes #9 = { convergent inlinehint norecurse nounwind "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" }
+attributes #10 = { convergent "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" }
+attributes #11 = { convergent mustprogress norecurse "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "sycl-module-id"="C:/Users/psamolys/Dev/sycl/libdevice/fallback-cassert.cpp" }
+attributes #12 = { nounwind }
+attributes #13 = { convergent }
+attributes #14 = { convergent nounwind }
+
+!llvm.dependent-libraries = !{!0, !0, !0}
+!opencl.spir.version = !{!1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1}
+!spirv.Source = !{!2, !2, !2, !2, !2, !2, !2, !2, !2, !2, !2, !2}
+!llvm.ident = !{!3, !4, !4, !4, !4, !4, !4, !4, !4, !4, !4, !4}
+!llvm.module.flags = !{!5, !6}
+
+!0 = !{!"libcpmt"}
+!1 = !{i32 1, i32 2}
+!2 = !{i32 4, i32 100000}
+!3 = !{!"clang version 14.0.0 (https://github.com/intel/llvm.git bb0055ccd006e861d781dc6332df658364590350)"}
+!4 = !{!"clang version 14.0.0 (https://github.com/intel/llvm.git 23e180bce8024817348ea07b1f5eefe15c7ace32)"}
+!5 = !{i32 1, !"wchar_size", i32 2}
+!6 = !{i32 7, !"frame-pointer", i32 2}
+!7 = !{i32 -1, i32 -1, i32 -1, i32 -1}
+!8 = !{!9, !9, i64 0}
+!9 = !{!"any pointer", !10, i64 0}
+!10 = !{!"omnipotent char", !11, i64 0}
+!11 = !{!"Simple C++ TBAA"}
+!12 = !{!10, !10, i64 0}
+!13 = !{!14, !14, i64 0}
+!14 = !{!"int", !10, i64 0}
+!15 = !{!16, !16, i64 0}
+!16 = !{!"long long", !10, i64 0}
+!17 = distinct !{!17, !18, !19}
+!18 = !{!"llvm.loop.mustprogress"}
+!19 = !{!"llvm.loop.unroll.enable"}
+!20 = !{i32 -1}
+!21 = !{i64 0, i64 8, !8}
+!22 = !{!23, !9, i64 0}
+!23 = !{!"_ZTSN2cl4sycl14kernel_handlerE", !9, i64 0}
+!24 = !{!25, !14, i64 0}
+!25 = !{!"_ZTS14AssertHappened", !14, i64 0, !10, i64 4, !10, i64 261, !10, i64 518, !14, i64 648, !16, i64 656, !16, i64 664, !16, i64 672, !16, i64 680, !16, i64 688, !16, i64 696}
+!26 = distinct !{!26, !18}
+!27 = !{i64 0, i64 4, !13, i64 4, i64 257, !12, i64 261, i64 257, !12, i64 518, i64 129, !12, i64 648, i64 4, !13, i64 656, i64 8, !15, i64 664, i64 8, !15, i64 672, i64 8, !15, i64 680, i64 8, !15, i64 688, i64 8, !15, i64 696, i64 8, !15}
```

Comment:
I suggest simplifying the test a bit - many of these metadata and attributes are not needed to check spec constants instructions.
And it'd be nice to remove unused type definitions at the beginning of the test (e.g., `%"struct.cl::sycl::detail::AssertHappened"`), it'll make the test more "unit". Also you can remove some unrelated instructions from the test function itself (like `@llvm.lifetime` intrinsic) - they should not affect the tested change).

Reply:
@vmaksimo Thank you for the suggestions. I've simplified the test a bit.